test/parallel/test-path.js

```diff
@@ -15,6 +15,12 @@ assert.equal(path.basename('/basename.ext'), 'basename.ext');
 assert.equal(path.basename('basename.ext'), 'basename.ext');
 assert.equal(path.basename('basename.ext/'), 'basename.ext');
 assert.equal(path.basename('basename.ext//'), 'basename.ext');
+assert.equal(path.basename('aaa/bbb', '/bbb'), 'bbb');
+assert.equal(path.basename('aaa/bbb', 'a/bbb'), 'bbb');
+assert.equal(path.basename('aaa/bbb', 'bbb'), 'bbb');
+assert.equal(path.basename('aaa/bbb//', 'bbb'), 'bbb');
+assert.equal(path.basename('aaa/bbb', 'bb'), 'b');
+assert.equal(path.basename('aaa/bbb', 'b'), 'bb');
```

Comment:
I was thinking maybe something more like `path.basename('/aaa/bbb')` since we have seen regressions for absolute vs relative paths. They may already be there though